### PR TITLE
Add controllers

### LINKED ISSUE #3: Add controllers
In this task need to add controllers to the project.

### PR DESCRIPTION
**Pull request by Eshaik**
Building controllers of Todo and Items with Test Driven Development (TDD), also adding more tests and some gems.

closes #3

### CODE REVIEW

**Comment by Eshaik** [comment]
@juanhurtado10 I did the rebase in this branch too!

File: spec/requests/todos_spec.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      end
+    end
+  end
+
+  # Test suite for PUT /todos/:id
+  describe 'PUT /todos/:id' do
+    let(:valid_attributes) { { title: 'Shopping' } }
+
+    context 'when the record exists' do
+      before { put "/todos/#{todo_id}", params: valid_attributes }
```
Comment (juanhurtado10) [inline spec/requests/todos_spec.rb:88]: @Eshaik Definitelly you should use the Routes helpers here

Comment (juanhurtado10) [reply spec/requests/todos_spec.rb:88]: Also for all the other code instances that require a route!

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: spec/requests/todos_spec.rb (hunk 1 above)
Comment (Eshaik) [reply spec/requests/todos_spec.rb]: @juanhurtado10 Oops, my bad. Now is using Routes helpers!

File: spec/requests/items_spec.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+      it 'returns a not found message' do
+        expect(response.body).to match(/Couldn't find Item/)
+      end
+    end
+  end
+
+  # Test suite for DELETE /todos/:id
+  describe 'DELETE /todos/:id' do
+    before { delete "/todos/#{todo_id}/items/#{id}" }
```
Comment (juanhurtado10) [inline spec/requests/items_spec.rb:121]: @Eshaik same case here! You're using concatenated strings for routes

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: spec/requests/items_spec.rb (hunk 2 above)
Comment (Eshaik) [reply spec/requests/items_spec.rb]: @juanhurtado10 I'm sorry. Now is using Routes helpers in `items_spec.rb` too!

**Review by juanhurtado10 — APPROVED** [review]

**Pull request merged**